scipy/integrate/_tanhsinh.py

```diff
@@ -324,8 +321,8 @@ def _tanhsinh(f, a, b, *, args=(), log=False, maxfun=None, maxlevel=None,
         c[inf_a] = b[inf_a]  # takes care of infinite a
         c[inf_b] = a[inf_b]  # takes care of infinite b
```

Comment:
I know I just changed these lines, but it needs one more tweak:
An advantage of the tanh-sinh scheme is its relative insensitivity to the function behavior at either limit of integration. However, for this initialization, we currently use the finite limit of integration to test evaluation of the callable if exactly one of the limits is infinite. This can cause a problem if the callable returns NaN at the limit of integration. For example, without special care, this is the behavior of the [log-laplace pdf](https://en.wikipedia.org/wiki/Log-Laplace_distribution).

```suggestion
        c[inf_a] = b[inf_a] - 1  # takes care of infinite a
        c[inf_b] = a[inf_b] + 1  # takes care of infinite b
```

Reply:
Looks reasonable. Go ahead and commit this.